transport/serialize/serializer_test.go

```diff
@@ -87,6 +175,49 @@ func TestJSONDeserialize(t *testing.T) {
 	if !reflect.DeepEqual(msg, expect) {
 		t.Fatalf("got %+v, expected %+v", msg, expect)
 	}
+
+	emptyData := `[]`
+	_, err = s.Deserialize([]byte(emptyData))
+	if err == nil {
+		t.Fatal("Empty message should be errored while decoding")
+	}
+}
+
+func TestJSONSerializeDataItem(t *testing.T) {
+	s := &JSONSerializer{}
+	b, err := s.SerializeDataItem(dataItem)
+	if err != nil {
+		t.Fatal("Serialization error: ", err)
+	}
+	if len(b) == 0 {
+		t.Fatal("no serialized data")
+	}
+
+	var res interface{}
+	if err := s.DeserializeDataItem(b, &res); err != nil {
+		t.Fatal("desrialization error: ", err)
+	}
+
+	compareDeserializedSerializedDataItem(t, res)
+}
+
+func TestJSONDeserializeDataItem(t *testing.T) {
+	s := &JSONSerializer{}
+
+	data := `[{"Arguments":[1,"2",true],"ArgumentsKw":{"prop1":1,"prop2":"2","prop3":true}}]`
+	var msg interface{}
+	if err := s.DeserializeDataItem([]byte(data), &msg); err != nil {
+		t.Fatalf("Error decoding good data: %s, %s", err, data)
+	}
+	compareDeserializedSerializedDataItem(t, msg)
+
+	castTo := &[]struct {
+		Arguments   wamp.List
+		ArgumentsKw wamp.Dict
+	}{}
+	if err := s.DeserializeDataItem([]byte(data), &castTo); err != nil {
+		t.Fatalf("Error decoding good data: %s, %s", err, data)
+	}
```

Comment:
You should also verify that what got deserialized is what is expected, eg:
```
	t.Logf("%+v", castTo)

	expected := []struct {
		Arguments   wamp.List
		ArgumentsKw wamp.Dict
	}{
		{
			Arguments: []interface{}{
				uint64(1), "2", true,
			},
			ArgumentsKw: map[string]interface{}{
				"prop1": uint64(1), "prop2": "2", "prop3": true,
			},
		},
	}
	eq := reflect.DeepEqual(castTo, expected)
	if !eq {
		t.Fatalf("should be true, expected:\n%v\ngot:\n%v",
			spew.Sdump(expected), spew.Sdump(castTo))

	}
```

Reply:
yeah. Done. 
Unfortunately have to copy/past check to every serializer test instead of creating helper like `compareDeserializedSerializedDataItem` because messagepack serializer decode `1` to `int64` instead of `uint64` so DeepEqual fails.